src/Fetchers/TMSFetcher/TMSFetcher.ts

```diff
@@ -139,9 +143,39 @@ export class TMSFetcher extends Fetcher {
      */
     descriptionToAppend(competition: Competition, match: Match,
                         html: boolean): string[] {
-
         const lines: string[] = [];
 
+        // Add officials if present
+        const officials = match.getOfficials();
+        if (officials.length > 0) {
+            lines.push("");
+            lines.push("Match Officials:");
+
+            // Define role order based on table columns
+            const roleOrder = [
+                "Umpire",
+                "Reserve/Video",
+                "Scoring/Timing",
+                "Technical Officer"
+            ];
+
+            // Sort officials by predefined role order
+            const sortedOfficials = [...officials].sort((a, b) => {
+                const indexA = roleOrder.indexOf(a.role) ?? 0;
```

Comment:
indexOf takes O(n) time, which is executed each `sort` iteration for every official for every match. The current implementation is therefore quite slow.

Instead of an array solution, please create an object like this:

```js
const roleOrder = {
  "Umpire": 0,
  "Reserve/Video": 1,
   ...
}
```

and access it using roleOrder[a.role]. This will drastically improve running time.